client/src/app/common/CustomRules/rules-utils.tsx

```diff
@@ -44,3 +44,10 @@ export const parseRules = (file: IReadFile): ParsedRule => {
     };
   }
 };
+
+export const getruleBundleTargetList = (ruleBundle: RuleBundle) => {
+  return ruleBundle.rulesets.reduce((acc: string[], ruleset) => {
+    acc.push(ruleset?.metadata?.target);
+    return acc;
```

Comment:
I think this is probably okay, but technically I don't think it's guaranteed to be safe to mutate the accumulator in-place in a reduce like this. You could do it immutably with a spread operator:
```ts
return [...acc, ruleset?.metadata?.target];
```